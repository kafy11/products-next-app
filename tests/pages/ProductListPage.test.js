import { shallow } from 'enzyme';
import ProductListPage from '../../pages/index';
import products from '../fixtures/products';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<ProductListPage products={products} />);
});

describe('ProductListPage', () => {
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should pass the products to the ProductTable', () => {
        const productsProp = wrapper.find('ProductTable').prop('products');
        expect(productsProp).toEqual(products);
    });
});