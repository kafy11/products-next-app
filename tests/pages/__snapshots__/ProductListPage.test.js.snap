// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListPage should render correctly 1`] = `
<Page
  title="Product List"
  toolbarContent={
    <LinkButton
      className="float-right"
      href="/add"
      title="Add Product"
    >
      <i
        className="fas fa-plus"
      />
    </LinkButton>
  }
>
  <ProductTable
    onDeleteProduct={[Function]}
    products={
      Array [
        Object {
          "name": "PS4",
          "price": 399,
          "product_id": 1,
          "quantity": 1,
        },
        Object {
          "name": "Xbox One",
          "price": 499,
          "product_id": 2,
          "quantity": 2,
        },
        Object {
          "name": "Nintendo Switch",
          "price": 299,
          "product_id": 3,
          "quantity": 3,
        },
      ]
    }
  />
</Page>
`;
